packages/container/lib/registry.js

```diff
@@ -692,9 +694,13 @@ Registry.prototype = {
     return assign({}, fallbackKnown, localKnown, resolverKnown);
   },
 
-  validateFullName(fullName) {
+  validateFullName(fullName, tolerant) {
```

Comment:
How about this alternative: 

* Create a new method `isValidFullName` that returns a boolean and never throws. 
* `validateFullName` could internally call `isValidFullName`, as could `has`.
* `validateFullName` could continue to throw.

I think this works because `validateFullName` sounds imperative and I would expect it to throw, plus the `tolerant` param seems a bit iffy.